src/probability/conditional.lean

```diff
@@ -0,0 +1,134 @@
+ensures that this is a probability measure (when `μ` is a finite measure).
+
+From this definition, we derive the "axiomatic" definition of conditional probability
+based on application: for any `s t : set α`, we have `μ[t|s] = (μ s)⁻¹ * μ (s ∩ t)`.
+
+## Main Statements
+
+* `cond_cond_eq_cond_inter`: conditioning on one set and then another is equivalent
+  to conditioning on their intersection.
+* `bayes`: Bayes' Theorem.
+
+## Notations
+
+This file uses the local notation `μ[|s]` the measure of `μ` conditioned on `s`,
+and `μ[t|s]` for the probability of `t` given `s` under `μ` (equivalent to the
+application `μ[|s] t`).
+
+## Implementation notes
+
+Because we have the alternative measure restriction application principles
+`measure.restrict_apply` and `measure.restrict_apply'`, which require 
+measurability of the restricted and restricting sets, respectively,
+many of the theorems here will have corresponding alternatives as well.
+For the sake of brevity, we've chosen to only go with `measure.restrict_apply'`
+for now, but the alternative theorems can be added if needed.
+
+Simplification generally follows the rule of removing conditions on a measure
+when possible.
+
+## Tags
+conditional, conditioned, bayes
+-/
+
+noncomputable theory
+
+open measure_theory measurable_space
+
+variables {α : Type*} [m : measurable_space α] (μ : measure α)
+
+namespace probability_theory
+
+section definitions
+
+/-- The conditional probability measure of measure `μ` on set `s` is `μ` restricted to `s` 
+and scaled by the inverse of `μ s` (to make it a probability measure):
+`(μ s)⁻¹ • μ.restrict s`. -/
+def cond (s : set α) : measure α :=
+  (μ s)⁻¹ • μ.restrict s
+
+end definitions
+
+local notation  μ `[` s `|` t `]` := cond μ t s
+local notation  μ `[|`:60 t`]` := cond μ t
+
+/-- The conditional probability measure of any finite measure on any set of positive measure
+is a probability measure. -/
+lemma cond_is_probability_measure [is_finite_measure μ] {s : set α} (hcs : μ s ≠ 0) :
+  is_probability_measure $ μ[|s] :=
+⟨by { rw [cond, measure.smul_apply, measure.restrict_apply measurable_set.univ,
+  set.univ_inter], exact ennreal.inv_mul_cancel hcs (measure_ne_top _ s) }⟩
+
+section bayes
+
+@[simp] lemma cond_univ [is_probability_measure μ] : μ[|set.univ] = μ :=
+by simp [cond, measure_univ, measure.restrict_univ]
+
+/-- The axiomatic definition of conditional probability derived from a measure-theoretic one. -/
+@[simp] lemma cond_measure_apply {s : set α} (hms : measurable_set s) (t : set α) :
+  μ[t|s] = (μ s)⁻¹ * μ (s ∩ t) :=
+by { rw [cond, measure.smul_apply, measure.restrict_apply' hms, set.inter_comm], refl }
+
+lemma inter_pos_of_cond_pos {s t : set α} (hms : measurable_set s)
+  (hcst : μ[t|s] ≠ 0) : μ (s ∩ t) ≠ 0 :=
+begin
+  refine right_ne_zero_of_mul _, exact (μ s)⁻¹,
+  convert hcst,
+  simp [hms, set.inter_comm]
+end
+
+variable [is_finite_measure μ]
+
+lemma cond_pos_of_inter_pos {s t : set α} (hms : measurable_set s)
```

Comment:
Same remark about the name as before. Use `pos` for `0 <` and `ne_zero` for what you have here. Our general convention for ennreal is to use ne_zero in hypotheses and pos in conclusions.